[Bug, merge_df] ValueError: `how` must be one of {'inner', 'left', ..}, got full

In terminal:
```console
pdr_backend/aimodel/test/test_aimodel_data_factory.py::test_create_xy_reg__1exchange_1coin_1signal
```

Returns:
```text
================================================================================================= test session starts ==================================================================================================
platform darwin -- Python 3.12.3, pytest-8.2.1, pluggy-1.5.0
rootdir: /Users/trentmc/code/pdr-backend
configfile: pytest.ini
plugins: web3-6.19.0, dash-2.17.0, typeguard-4.2.1, env-1.1.3, asyncio-0.21.1, requests-mock-1.12.1
asyncio: mode=Mode.STRICT
collected 1 item                                                                                                                                                                                                       

pdr_backend/aimodel/test/test_aimodel_data_factory.py F                                                                                                                                                          [100%]

======================================================================================================= FAILURES =======================================================================================================
_____________________________________________________________________________________ test_create_xy_reg__1exchange_1coin_1signal ______________________________________________________________________________________

    @enforce_types
    def test_create_xy_reg__1exchange_1coin_1signal():
        feedset_list = [
            {
                "predict": "binanceus ETH/USDT h 5m",
                "train_on": "binanceus ETH/USDT h 5m",
            }
        ]
        d = predictoor_ss_test_dict(feedset_list=feedset_list)
        predictoor_ss = PredictoorSS(d)
        predict_feed = predictoor_ss.predict_train_feedsets[0].predict
        train_feeds = predictoor_ss.predict_train_feedsets[0].train_on
        aimodel_data_factory = AimodelDataFactory(predictoor_ss)
>       mergedohlcv_df = merge_rawohlcv_dfs(ETHUSDT_RAWOHLCV_DFS)

pdr_backend/aimodel/test/test_aimodel_data_factory.py:119: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/lib/python3.12/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
pdr_backend/lake/merge_df.py:38: in merge_rawohlcv_dfs
    merged_df = _add_df_col(merged_df, merged_col, raw_df, raw_col)
venv/lib/python3.12/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
pdr_backend/lake/merge_df.py:76: in _add_df_col
    merged_df = merged_df.join(newraw_df, on="timestamp", how="full")  # type: ignore
venv/lib/python3.12/site-packages/polars/dataframe/frame.py:6404: in join
    .join(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <LazyFrame at 0x168FB2B10>, other = <LazyFrame at 0x168A07B00>, on = 'timestamp', how = 'full'

    def join(
        self,
        other: LazyFrame,
        on: str | Expr | Sequence[str | Expr] | None = None,
        how: JoinStrategy = "inner",
        *,
        left_on: str | Expr | Sequence[str | Expr] | None = None,
        right_on: str | Expr | Sequence[str | Expr] | None = None,
        suffix: str = "_right",
        validate: JoinValidation = "m:m",
        join_nulls: bool = False,
        coalesce: bool | None = None,
        allow_parallel: bool = True,
        force_parallel: bool = False,
    ) -> Self:
        """
        Add a join operation to the Logical Plan.
    
        Parameters
        ----------
        other
            Lazy DataFrame to join with.
        on
            Join column of both DataFrames. If set, `left_on` and `right_on` should be
            None.
        how : {'inner', 'left', 'outer', 'semi', 'anti', 'cross', 'outer_coalesce'}
            Join strategy.
    
            * *inner*
                Returns rows that have matching values in both tables
            * *left*
                Returns all rows from the left table, and the matched rows from the
                right table
            * *outer*
                 Returns all rows when there is a match in either left or right table
            * *cross*
                 Returns the Cartesian product of rows from both tables
            * *semi*
                 Filter rows that have a match in the right table.
            * *anti*
                 Filter rows that not have a match in the right table.
    
            .. note::
                A left join preserves the row order of the left DataFrame.
        left_on
            Join column of the left DataFrame.
        right_on
            Join column of the right DataFrame.
        suffix
            Suffix to append to columns with a duplicate name.
        validate: {'m:m', 'm:1', '1:m', '1:1'}
            Checks if join is of specified type.
    
                * *many_to_many*
                    “m:m”: default, does not result in checks
                * *one_to_one*
                    “1:1”: check if join keys are unique in both left and right datasets
                * *one_to_many*
                    “1:m”: check if join keys are unique in left dataset
                * *many_to_one*
                    “m:1”: check if join keys are unique in right dataset
    
            .. note::
    
                - This is currently not supported the streaming engine.
        join_nulls
            Join on null values. By default null values will never produce matches.
        coalesce
            Coalescing behavior (merging of join columns).
            - None: -> join specific.
            - True: -> Always coalesce join columns.
            - False: -> Never coalesce join columns.
        allow_parallel
            Allow the physical plan to optionally evaluate the computation of both
            DataFrames up to the join in parallel.
        force_parallel
            Force the physical plan to evaluate the computation of both DataFrames up to
            the join in parallel.
    
        See Also
        --------
        join_asof
    
        Examples
        --------
        >>> lf = pl.LazyFrame(
        ...     {
        ...         "foo": [1, 2, 3],
        ...         "bar": [6.0, 7.0, 8.0],
        ...         "ham": ["a", "b", "c"],
        ...     }
        ... )
        >>> other_lf = pl.LazyFrame(
        ...     {
        ...         "apple": ["x", "y", "z"],
        ...         "ham": ["a", "b", "d"],
        ...     }
        ... )
        >>> lf.join(other_lf, on="ham").collect()
        shape: (2, 4)
        ┌─────┬─────┬─────┬───────┐
        │ foo ┆ bar ┆ ham ┆ apple │
        │ --- ┆ --- ┆ --- ┆ ---   │
        │ i64 ┆ f64 ┆ str ┆ str   │
        ╞═════╪═════╪═════╪═══════╡
        │ 1   ┆ 6.0 ┆ a   ┆ x     │
        │ 2   ┆ 7.0 ┆ b   ┆ y     │
        └─────┴─────┴─────┴───────┘
        >>> lf.join(other_lf, on="ham", how="outer").collect()
        shape: (4, 5)
        ┌──────┬──────┬──────┬───────┬───────────┐
        │ foo  ┆ bar  ┆ ham  ┆ apple ┆ ham_right │
        │ ---  ┆ ---  ┆ ---  ┆ ---   ┆ ---       │
        │ i64  ┆ f64  ┆ str  ┆ str   ┆ str       │
        ╞══════╪══════╪══════╪═══════╪═══════════╡
        │ 1    ┆ 6.0  ┆ a    ┆ x     ┆ a         │
        │ 2    ┆ 7.0  ┆ b    ┆ y     ┆ b         │
        │ null ┆ null ┆ null ┆ z     ┆ d         │
        │ 3    ┆ 8.0  ┆ c    ┆ null  ┆ null      │
        └──────┴──────┴──────┴───────┴───────────┘
        >>> lf.join(other_lf, on="ham", how="left").collect()
        shape: (3, 4)
        ┌─────┬─────┬─────┬───────┐
        │ foo ┆ bar ┆ ham ┆ apple │
        │ --- ┆ --- ┆ --- ┆ ---   │
        │ i64 ┆ f64 ┆ str ┆ str   │
        ╞═════╪═════╪═════╪═══════╡
        │ 1   ┆ 6.0 ┆ a   ┆ x     │
        │ 2   ┆ 7.0 ┆ b   ┆ y     │
        │ 3   ┆ 8.0 ┆ c   ┆ null  │
        └─────┴─────┴─────┴───────┘
        >>> lf.join(other_lf, on="ham", how="semi").collect()
        shape: (2, 3)
        ┌─────┬─────┬─────┐
        │ foo ┆ bar ┆ ham │
        │ --- ┆ --- ┆ --- │
        │ i64 ┆ f64 ┆ str │
        ╞═════╪═════╪═════╡
        │ 1   ┆ 6.0 ┆ a   │
        │ 2   ┆ 7.0 ┆ b   │
        └─────┴─────┴─────┘
        >>> lf.join(other_lf, on="ham", how="anti").collect()
        shape: (1, 3)
        ┌─────┬─────┬─────┐
        │ foo ┆ bar ┆ ham │
        │ --- ┆ --- ┆ --- │
        │ i64 ┆ f64 ┆ str │
        ╞═════╪═════╪═════╡
        │ 3   ┆ 8.0 ┆ c   │
        └─────┴─────┴─────┘
        """
        if not isinstance(other, LazyFrame):
            msg = f"expected `other` join table to be a LazyFrame, not a {type(other).__name__!r}"
            raise TypeError(msg)
    
        if how == "cross":
            return self._from_pyldf(
                self._ldf.join(
                    other._ldf,
                    [],
                    [],
                    allow_parallel,
                    force_parallel,
                    join_nulls,
                    how,
                    suffix,
                    validate,
                )
            )
    
        if on is not None:
            pyexprs = parse_as_list_of_expressions(on)
            pyexprs_left = pyexprs
            pyexprs_right = pyexprs
        elif left_on is not None and right_on is not None:
            pyexprs_left = parse_as_list_of_expressions(left_on)
            pyexprs_right = parse_as_list_of_expressions(right_on)
        else:
            msg = "must specify `on` OR `left_on` and `right_on`"
            raise ValueError(msg)
    
        if how == "outer_coalesce":
            coalesce = True
    
        return self._from_pyldf(
>           self._ldf.join(
                other._ldf,
                pyexprs_left,
                pyexprs_right,
                allow_parallel,
                force_parallel,
                join_nulls,
                how,
                suffix,
                validate,
                coalesce,
            )
        )
E       ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full

venv/lib/python3.12/site-packages/polars/lazyframe/frame.py:3992: ValueError
=============================================================================================== short test summary info ================================================================================================
FAILED pdr_backend/aimodel/test/test_aimodel_data_factory.py::test_create_xy_reg__1exchange_1coin_1signal - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
================================================================================================== 1 failed in 0.68s ===================================================================================================
```

----

Running test_merge_df.py also causes many failures. From console:
```console
pytest pdr_backend/lake/test/test_merge_df.py 
```

Returns:
```text
FAILED pdr_backend/lake/test/test_merge_df.py::test_mergedohlcv_df_shape - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
FAILED pdr_backend/lake/test/test_merge_df.py::test_merge_rawohlcv_dfs_equal_dfs - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
FAILED pdr_backend/lake/test/test_merge_df.py::test_merge_rawohlcv__empty_and_nonempty_df - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
FAILED pdr_backend/lake/test/test_merge_df.py::test_merge_rawohlcv__dfs_with_different_timestamps - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
FAILED pdr_backend/lake/test/test_merge_df.py::test_add_df_col_unequal_dfs - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
FAILED pdr_backend/lake/test/test_merge_df.py::test_add_df_col_equal_dfs - ValueError: `how` must be one of {'inner', 'left', 'outer', 'semi', 'anti', 'cross'}, got full
```

--- a/pdr_backend/lake/merge_df.py
+++ b/pdr_backend/lake/merge_df.py
@@ -73,7 +73,7 @@ def _add_df_col(
     if merged_df is None:
         merged_df = newraw_df
     else:
-        merged_df = merged_df.join(newraw_df, on="timestamp", how="full")
+        merged_df = merged_df.join(newraw_df, on="timestamp", how="outer")
         merged_df = merge_cols(merged_df, "timestamp", "timestamp_right")
 
     # re-order merged_df's columns

